whylog/constraints/__init__.py

```diff
@@ -0,0 +1,36 @@
+class Constraint(object):
```

Comment:
AbstractConstraint? It should define check() with an ABC decorator (arguments are TBD)